db/db_impl.cc

```diff
@@ -3054,5 +3061,68 @@ void DBImpl::WaitForIngestFile() {
   }
 }
 
+Status DBImpl::StartTrace(const TraceOptions& options,
+                          const std::string& trace_filename) {
```

Comment:
As discussed offline, it feels cleaner and if we only keep the StartTrace() which takes trace_writer. What is done here can be done in a factory function.